src/runtime_src/driver/xclng/tools/xbutil/xbutil.cpp

```diff
@@ -1479,3 +1493,174 @@ int xcldev::xclP2p(int argc, char *argv[])
 
     return ret;
 }
+
+/*
+ * m2mtest
+ */
+static void m2m_free_unmap_bo(xclDeviceHandle handle, unsigned boh, 
+auto * &boptr, size_t boSize)
+{
+    if(boh != NULLBO)
+        xclFreeBO(handle, boh);
+    if(boptr != nullptr)
+        munmap(boptr, boSize);
+}
+
+static int m2m_alloc_init_bo(xclDeviceHandle handle, char * &boptr, 
+size_t boSize, int bank, char pattern)
```

Comment:
indent here. you probably don't want to mix bo handle and error code as return values.